src/utils.rs

```diff
@@ -189,6 +189,40 @@ macro_rules! try_opt {
     })
 }
 
+pub fn wrap_str<S>(s: S, max_width: usize, width: usize, offset: usize) -> Option<String>
+    where S: Into<String>
```

Comment:
I think it might be better to take an `&str` rather than use `Into<String>` - it will save an allocation in some cases.